### DISCUSSION TITLE
ffmpeg on google colab.

### DISCUSSION THREAD

**Discussion by vitorpsogarcia** [AlUlkesh/sd_save_intermediate_images#28] [Q&A]
Hello, I've been running your extension through google colab, it saves the intermediates, but not the videos, any idea how I can use ffmpeg in the colab so that it is recognized by the intermediate images?

**Comment by AlUlkesh** [comment]
Have you tried
`!apt install ffmpeg`

?

**Reply by vitorpsogarcia** [reply]
Yes.

**Reply by AlUlkesh** [reply]
So, what does not work? Are there error messages? Did you try turning on the debug option of this extension?

**Reply by vitorpsogarcia** [reply]
![image](https://user-images.githubusercontent.com/65982758/226111869-2c1f05ea-20a9-457e-af02-f219e9196226.png)
![image](https://user-images.githubusercontent.com/65982758/226112859-22b5a919-094e-42c3-b8ed-dca60d76a7cd.png)
![image](https://user-images.githubusercontent.com/65982758/226112961-847004a4-a0a5-4173-863b-a6d42ebf9cc9.png)

This is what appears on colab

**Reply by AlUlkesh** [reply]
Thanks. It's an error on os.link. Apparently this is the case:

`If you connected to an SMB/FTP/.. (etc) file systems remotely, you can't create a hard link (security policy).`

(https://askubuntu.com/questions/1178797/ln-failed-to-create-symbolic-link-function-not-implemented)

I'll see if I can implement an alternative for this case.

**Reply by vitorpsogarcia** [reply]
First I want to thank you for the extension, it's amazing, although it doesn't work in Colab, I could already see that it's excellent by videos and the examples on GitHub, in fact, they served to make me even more eager to use it.

I'm looking forward to the update!

**Reply by AlUlkesh** [reply]
Thank you.

I first tried symobolic links, but they run into the same issue. Now I simply copy the files, if the link fails. That will temporarily take up more space, but it shouldn't be that much.

Can you check if it works for you with the latest update?

**Reply by vitorpsogarcia** [reply]
It fit like a glove. You just made me the happiest person on the face of planet earth.

https://drive.google.com/file/d/17ZUfaoixo3pGFll6tSAsbt4hmSa5uIW3/view?usp=sharing

**Reply by AlUlkesh** [reply]
Yay :smiley:
